haystack/document_store/faiss.py

```diff
@@ -455,17 +473,38 @@ def load(
 
         :param faiss_file_path: Stored FAISS index file. Can be created via calling `save()`
         :param sql_url: Connection string to the SQL database that contains your docs and metadata.
+            Overrides the value defined in the `faiss_init_params_path` file, if present
         :param index: Index name to load the FAISS index as. It must match the index name used for
-                      when creating the FAISS index.
-        :return:
-        """
-        """
+                      when creating the FAISS index. Overrides the value defined in the 
+                      `faiss_init_params_path` file, if present
+        :param faiss_init_params_path: Stored FAISS init parameters. Can be created via calling `save()`
+        :return: the DocumentStore
         """
+        if not faiss_init_params_path:
+            faiss_file_path = Path(faiss_file_path)
+            faiss_init_params_path = faiss_file_path.with_suffix(".json")
+
+        with open(faiss_init_params_path, 'r') as ipp:
```

Comment:
There is a case in which I'm unsure how is best to behave. What if the file does not exist because the user wants a FAISS index with all the default values? I think there are two "legitimate" behaviors here:

1) We strictly require this file for the method to work: if you want defaults, create an empty file and pass it in.
2) We don't require this file: if the file is not found, it silently falls back to the defaults, or emits a warning but then keeps going.

I don't like silent (or warning-based) fails in general, but in this case it might be easier than requiring everybody to create empty files at the side of their existing saved FAISS indexes... even though `save` would create them in most cases. I'm not convinced of what's best here. Opinions?

Reply:
Good question! 
If we had many cases where users store FAISS indices outside of Haystack and want to then load them, I would have probably voted for option 2. 
However, in our case, we can only load indices that have been created in Haystack as we also need the associated data in SQL (meta, texts ...) and this is hardly something users will generate outside of Haystack. That's why there won't bee many cases where users have to create an empty file. Only those who saved an index in an old Haystack version and now migrate to the newer one. For them we can add an info in the exception or docstring.

Reply:
Agreed!